nvtabular/workflow/node.py

```diff
@@ -252,22 +342,44 @@ def parents_schema(self):
 
     @property
     def input_columns(self):
+        if not self.input_schema:
+            raise RuntimeError(
+                "The input columns aren't computed until the workflow "
+                "is fit to a dataset or input schema."
+            )
+
         if self.selector:
+            # To maintain column groupings
             return self.selector
         else:
             return ColumnSelector(self.input_schema.column_names)
 
     @property
     def output_columns(self):
-        if self.op:
-            return self.op.output_column_names(self.input_columns)
-        else:
-            return self.input_columns
+        if not self.input_schema:
+            raise RuntimeError(
+                "The output columns aren't computed until the workflow "
+                "is fit to a dataset or input schema."
+            )
+
+        return ColumnSelector(self.output_schema.column_names)
 
     @property
     def dependency_columns(self):
         return ColumnSelector(self.dependencies_schema.column_names)
 
+    @property
+    def dependency_nodes(self):
+        nodes = []
+        for dep in self.dependencies:
+            if isinstance(dep, WorkflowNode):
+                nodes.append(dep)
+            elif isinstance(dep, list):
+                for nested_dep in dep:
+                    if isinstance(nested_dep, WorkflowNode):
+                        nodes.append(nested_dep)
```

Comment:
Same iterate through nested lists code from before, should be easier to compute than this